.github/workflows/dependabot-automerge.yaml

```diff
@@ -0,0 +1,50 @@
+name: dependabot merger
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '00 03 * * Mon' # trigger every Monday at 3:00 a.m., as our dependabot is configured to raise PRs every Monday at 2:00 a.m.
+
+env:
+  DEPENDABOT_GROUPS: |
+    production-minor-patch group
+    plugins group
+    test group
+    github-actions group
+jobs:
+  review-prs:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Approve and Merge PRs
+        run: |
+          PRS=$(gh pr list --app "dependabot" --state "open" --json number,title)
+          PR_NUMBERS=
+          while IFS= read -r GROUP; do
+            if [[ -z "$GROUP" ]]; then
+              continue
+            fi
+          
+            MATCHES=$(jq -r --arg group "$GROUP" '.[] | select(.title | contains($group)) | .number' <<< "$PRS")
+            echo "[DEBUG] Found PRs for group '$GROUP': '$MATCHES'"
+          
+            PR_NUMBERS="$MATCHES"$'\n'"$PR_NUMBERS"
+          done <<< "${{ env.DEPENDABOT_GROUPS }}"
+          echo "[DEBUG] Approving and Merging following PRs: '$PR_NUMBERS'"
```

Comment:
How do you check for approving/merging only ✔️ successfully built PRs?

Reply:
Everything is approved, only green PRs are merge because of the rule we have in place